src/Microsoft.Health.Dicom.Core/Features/HealthCheck/BackgroundServiceHealthCheck.cs

```diff
@@ -1,87 +0,0 @@
-using Microsoft.Health.Dicom.Core.Features.Store;
-using Microsoft.Health.Dicom.Core.Features.Telemetry;
-using Microsoft.Health.Encryption.Customer.Health;
-
-namespace Microsoft.Health.Dicom.Core.Features.HealthCheck;
-
-public class BackgroundServiceHealthCheck : IHealthCheck
-{
-    private const string DegradedDescription = "The health of the background service has degraded.";
-
-    private readonly IIndexDataStore _indexDataStore;
-    private readonly DeletedInstanceCleanupConfiguration _deletedInstanceCleanupConfiguration;
-    private readonly DeleteMeter _deleteMeter;
-    private readonly BackgroundServiceHealthCheckCache _backgroundServiceHealthCheckCache;
-    private readonly ValueCache<CustomerKeyHealth> _customerKeyHealthCache;
-    private readonly ILogger<BackgroundServiceHealthCheck> _logger;
-
-    public BackgroundServiceHealthCheck(
-        IIndexDataStore indexDataStore,
-        IOptions<DeletedInstanceCleanupConfiguration> deletedInstanceCleanupConfiguration,
-        DeleteMeter deleteMeter,
-        BackgroundServiceHealthCheckCache backgroundServiceHealthCheckCache,
-        ValueCache<CustomerKeyHealth> customerKeyHealthCache,
-        ILogger<BackgroundServiceHealthCheck> logger)
-    {
-        EnsureArg.IsNotNull(indexDataStore, nameof(indexDataStore));
-        EnsureArg.IsNotNull(deletedInstanceCleanupConfiguration?.Value, nameof(deletedInstanceCleanupConfiguration));
-        EnsureArg.IsNotNull(deleteMeter, nameof(deleteMeter));
-        EnsureArg.IsNotNull(backgroundServiceHealthCheckCache, nameof(backgroundServiceHealthCheckCache));
-        EnsureArg.IsNotNull(customerKeyHealthCache, nameof(customerKeyHealthCache));
-        EnsureArg.IsNotNull(logger, nameof(logger));
-
-        _indexDataStore = indexDataStore;
-        _deletedInstanceCleanupConfiguration = deletedInstanceCleanupConfiguration.Value;
-        _deleteMeter = deleteMeter;
-        _backgroundServiceHealthCheckCache = backgroundServiceHealthCheckCache;
-        _customerKeyHealthCache = customerKeyHealthCache;
-        _logger = logger;
-    }
-
-    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
-    {
-        try
-        {
-            CustomerKeyHealth cmkStatus = await _customerKeyHealthCache.GetAsync(cancellationToken).ConfigureAwait(false);
-            if (!cmkStatus.IsHealthy)
-            {
-                // if the customer-managed key is inaccessible, the data store will also be inaccessible
-                return new HealthCheckResult(
-                    HealthStatus.Degraded,
-                    DegradedDescription,
-                    cmkStatus.Exception,
-                    new Dictionary<string, object> { { "Reason", cmkStatus.Reason } });
-            }
-
-            Task<DateTimeOffset> oldestWaitingToBeDeleted = _backgroundServiceHealthCheckCache.GetOrAddOldestTimeAsync(_indexDataStore.GetOldestDeletedAsync, cancellationToken);
```

Comment:
The only noticeable logic in this health check is the querying and logging of these metrics